e2e-chatbot-app-next/client/src/lib/pdf-utils.ts

```
@@ -0,0 +1,133 @@
+ *
+ * Expected URL format: /ajax-api/2.0/fs/files/Volumes/{catalog}/{schema}/{volume}/{filename}.pdf
+ * Optional hash: #page=N:~:text=encoded_text or #:~:text=encoded_text or #page=N
+ */
+export function parseUnityCatalogPDFLink(href: string): UCPDFMetadata | null {
+  try {
+    const urlObject = new URL(href, window.location.origin);
+    const pathname = decodeURIComponent(urlObject.pathname);
+
+    const regex =
+      /^\/ajax-api\/2\.0\/fs\/files\/(Volumes\/[^/]+\/[^/]+\/[^/]+)\/(.+\.pdf)$/i;
+    const match = pathname.match(regex);
+
+    if (match) {
+      let page: number | undefined = undefined;
+      let textFragment: string | undefined = undefined;
+
+      // Check if page is in query params (legacy format)
+      const pageParam = urlObject.searchParams.get('page');
+      if (pageParam) {
+        const parsedPage = Number.parseInt(pageParam, 10);
+        page = !Number.isNaN(parsedPage) ? parsedPage : undefined;
+      }
+
+      // Check if hash contains page and/or text fragment (new format)
+      // Hash can be: #page=5:~:text=cited%20text or just #:~:text=cited%20text or #page=5
+      if (urlObject.hash) {
+        const hash = urlObject.hash.substring(1); // Remove the '#' prefix
+
+        // Check for page in hash (format: page=5:~:text=... or page=5)
+        const hashPageMatch = hash.match(/^page=(\d+)/);
+        if (hashPageMatch) {
+          const parsedPage = Number.parseInt(hashPageMatch[1], 10);
+          page = !Number.isNaN(parsedPage) ? parsedPage : page;
+        }
+
+        // Check for text fragment in hash (format: :~:text=...)
+        const textFragmentMatch = hash.match(/:~:text=(.+)$/);
+        if (textFragmentMatch) {
+          textFragment = decodeURIComponent(textFragmentMatch[1]);
+        } else if (hash.startsWith(':~:text=')) {
+          // Handle case where hash only contains text fragment without page
+          textFragment = decodeURIComponent(hash.substring(':~:text='.length));
+        }
```

this only triggers when hash is exactly :~:text= (empty text fragment) because the regex .+ requires at least one character, right? would this ever occur in practice? should we throw an error something instead?